// gatsby-plugin-emotion/gatsby-node.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.onCreateBabelConfig = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var onCreateBabelConfig = function onCreateBabelConfig(_ref, pluginOptions) {
  var actions = _ref.actions;
  var pragmaName = "___EmotionJSX";
  actions.setBabelPlugin({
    name: "babel-plugin-jsx-pragmatic",
    options: {
      export: "jsx",
      module: "@emotion/core",
      import: pragmaName
    }
  });
  actions.setBabelPlugin({
    name: "@babel/plugin-transform-react-jsx",
    options: {
      pragma: pragmaName,
      pragmaFrag: "React.Fragment"
    }
  });
  actions.setBabelPlugin({
    name: "babel-plugin-emotion",
    options: (0, _extends2.default)({
      cssPropOptimization: true,
      sourceMap: process.env.NODE_ENV !== "production",
      autoLabel: process.env.NODE_ENV !== "production"
    }, pluginOptions ? pluginOptions : {})
  });
};

exports.onCreateBabelConfig = onCreateBabelConfig;
